study.md

```diff
@@ -21,18 +21,49 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+I am confused about nested callbacks and the example given in the article about return:
+
+function foo(){
+    function bar() {
+        return 3;
+    }
+    return bar();
+    function bar() {
+        return 8;
+    }
+}
+alert(foo());
+
+I understand that something about the way bar() is defined both times causes JavaScript to read it before the return but I'm not sure what that thing is.
 ```
 
 Are there any nagging questions relating to functions that you have?
 
 ```md
-<!-- your answer here -->
+What is the difference between the various ways of declaring functions? we touched on the
+difference between the arrow notation and typing out a function literal but there are
+apparently noteworthy differences between
+
+function aFunction() {
+  return true;
+}
+
+and
+
+const aFunction = function () {
+  return true;
+};
+
+I would think (assuming I do not have a syntax error somewhere in there) that these should
```

Comment:
I wrote this short script and hopefully it will clear some things up.  If not, then let me know.
```javascript
// Function declaration.
function functionDeclaration() {
  console.log(this);
}

// Function expression.
const functionExpression = function() {
  console.log(this);
};

// Arrow function expression.  This binds "this" to its caller (the function)
// and returns an empty object.
const arrowFunctionExpression = () => {
  console.log(this);
};

// Returns the global object.
functionDeclaration();

// Returns the global object.
functionExpression();

// Returns an empty object, {}.
arrowFunctionExpression();
```